src/features/duckplayer/overlay-messages.js

```diff
@@ -12,19 +12,34 @@ import * as constants from './constants.js'
 export class DuckPlayerOverlayMessages {
     /**
      * @param {Messaging} messaging
+     * @param {ImportMeta['injectName']} injectName
      * @internal
      */
-    constructor (messaging) {
+    constructor (messaging, injectName) {
         /**
          * @internal
          */
         this.messaging = messaging
+        this.injectName = injectName
     }
 
     /**
      * @returns {Promise<import("../duck-player.js").OverlaysInitialSettings>}
      */
     initialSetup () {
+        if (this.injectName === 'integration') {
+            const overlayCopy = /** @type {import("../duck-player.js").UISettings['overlayCopy']} */ (new URLSearchParams(window.location.href).get('overlayCopy') || 'default')
+
+            return Promise.resolve({
+                userValues: {
+                    overlayInteracted: false,
+                    privatePlayerMode: { alwaysAsk: {} }
+                },
+                ui: {
+                    overlayCopy
+                }
+            })
+        }
```

Comment:
We need to avoid accessing the window url like this here. There's an example over here in the Environment class https://github.com/duckduckgo/content-scope-scripts/blob/main/src/features/duckplayer/overlays.js#L140-L160

the idea is that we use the `if (this.debug)` flag to guard against anything running in production. Also we keep all access to things like URLs/search params here too.

I'd prefer if the check here was purely in the Environment class, and it can be a combination of checks like injectName, debug etc.

Remember this code is injected into web pages, so we have different constraints/preferences here.

Reply:
I’ve moved that check into the Environment class and changed DuckPlayerOverlayMessages to accept an instance of Environment in the constructor